hs/src/Reach/AST/DK.hs

```diff
@@ -84,13 +84,16 @@ instance Pretty DKBlock where
   pretty (DKBlock _ _ k a) = prettyBlockP k a
 
 data DKProg
-  = DKProg SrcLoc DLOpts SLParts DLInit DKTail
+  = DKProg SrcLoc DLOpts SLParts DLInit [(SLVar, DLExportValue)] DKTail
```

Comment:
I think you should make a type like `type DLinExports a = M.Map SLVar (DLinExportVal a)` where the `a` is the type of the function body, then make a family of those for `DLExports`, `DKExports`, and so on